File: 9781430238010/9781430238010_Ch13/IfDef.java
/**
 * illustration of conditional compilation in Java
 * or - how to fake out the Java compiler
 * jfdooley 02/23/2011 (parts retrieved from the internet)
 */

public class IfDef {
	final static boolean DEBUG = true;

	public static void main(String [] args) {
		System.out.printf("Hello World \n");

		if (DEBUG) {
		    System.out.printf("max(5, 8) is %d\n", Math.max(5, 8));
			System.out.printf("If this prints, the code was included\n");
		}
	}
}

File: 9781430238010/9781430238010_Ch13/ifdef.c
#include <stdlib.h>
#include <stdio.h>

#define DEBUG 1
#define max(a, b) (a>b)?a:b


int main(int argc, char **argv)
{
	printf("Hello World\n");

#ifdef DEBUG
	printf("max(5, 8) is %d\n", max(5, 8));
	printf("This is a debug statement\n");
#endif

	return 0;
}
